### PR TITLE
Update disk.asm

### PR DESCRIPTION
**Pull request by Whammo**
use logical file number table to close_all_logical_files:

### CODE REVIEW

File: disk.asm [hunk 1]
```diff
@@ -301,12 +301,9 @@ INCLUDED
...
-    pla
-    tax
-    dex
-    bne -
-    rts
+-   ldx $98    ; number of open files
+    beq +       
+    lda $259,x ; logical file number table
+    jsr CLOSE  ; decrements $98 
+    bcc -
```
Comment (jkotlinski) [inline disk.asm:308]: Let's assume if CLOSE returns with carry set, it indicates an error.

Would it still make sense to close the remaining files?

I.e. replace `bcc -` with `jmp close_all_logical_files`?

Comment (jkotlinski) [reply disk.asm:308]: A small note: For C128, the logical file number table has a different location, so this code is not directly portable.
Instead, the kernal routine CLOSE_ALL could be used to close all open files.

Comment (Whammo) [reply disk.asm:308]: That's why you make the big bucks!  $$$

Comment (jkotlinski) [reply disk.asm:308]: ...wait. C128 CLOSE_ALL does not work like I thought, it works per-device. So a routine like this is still useful.

Comment (Whammo) [reply disk.asm]: They couldn't fit this into the Kernal.

**Comment by jkotlinski** [comment]
Sorry, I considered it but I'm not picking up this one. The old code is inefficient, but it doesn't really matter since it only runs in exceptional cases. So then I value more that it's portable.

**Pull request closed without merging**